File: peek_plugin_graphdb/_private/tuples/ItemKeyTypeTuple.py
from vortex.Tuple import Tuple, addTupleType, TupleField

from peek_plugin_graphdb._private.PluginNames import graphDbTuplePrefix


@addTupleType
class ItemKeyTypeTuple(Tuple):
    """ ItemKeyIndex Tuple

    This tuple is the publicly exposed ItemKeyIndex

    """
    __tupleType__ = graphDbTuplePrefix + 'ItemKeyTypeTuple'

    #:  A protected variable
    id__: int = TupleField()

    #:  The unique key of this itemKeyIndex
    key: str = TupleField()

    #:  The model set key of this itemKeyIndex
    modelSetKey: str = TupleField()

    #:  The unique title of itemKeyIndex
    name: str = TupleField()

File: peek_plugin_graphdb/plugin-module/_private/item-key-index-loader/ItemKeyTypeTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {graphDbTuplePrefix} from "./_private/PluginNames";


@addTupleType
export class ItemKeyTypeTuple extends Tuple {
    public static readonly tupleName = graphDbTuplePrefix + "ItemKeyTypeTuple";

    //  A protected variable
    id__: number;

    //  The key of this ItemKeyType
    key: string;

    //  The key of the model set
    modelSetKey: string;

    //  The name of the ItemKeyType
    name: string;

    constructor() {
        super(ItemKeyTypeTuple.tupleName)
    }
}